c7n/resources/sagemaker.py

```diff
@@ -661,6 +709,37 @@ def process(self, jobs):
                 pass
 
 
+@SagemakerLabelingJob.action_registry.register('stop')
+class SagemakerLabelingJobStop(SagemakerJobStop):
+    """Stops a SageMaker labeling job
+
+    :example:
+
+    .. code-block:: yaml
+
+        policies:
+          - name: stop-labeling-job
+            resource: aws.sagemaker-labeling-job
+            filters:
+              - LabelingJobName: label-job-10
+            actions:
+              - stop
+    """
+    permissions = ('sagemaker:StopLabelingJob',)
+    valid_origin_states = ('InProgress',)
+
+    def process(self, jobs):
+        jobs = self.filter_resources(jobs, 'LabelingJobStatus', self.valid_origin_states)
```

Comment:
what's the use case if its a transient resource to stop?

Reply:
Come to think of it, the use case I had in mind is actually pretty brittle. Was thinking of a scenario where we might see it is not using a KMS Key to encrypt the data and wanted to stop processing. Not sure how long that actual window would be though so this may be superflous. Should I take it out?

Reply:
it makes wonder about the use cases for all the labeling stuff (resource, filter, action)?. also how long do completed jobs stick around, does it need default server side filtering (like we do for emr clusters)?

Reply:
I checked with some of our proserv guys and they told me it is currently forever and they cannot be deleted.. which is interesting. I wonder what that starts to look like at a company that uses them all of the time :/

Reply:
The utility on the stop action is mostly around compat with the KMS Key check so I'd like to keep both in. Otherwise we get stuck with just notify/tag combos.